junit5/src/main/java/com/linecorp/armeria/testing/junit5/server/ServiceRequestContextCaptor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.testing.junit5.server;
+
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.function.Function;
+
+/**
+ * Captures the {@link ServiceRequestContext}s.
+ */
+public final class ServiceRequestContextCaptor {
+    private final BlockingQueue<ServiceRequestContext> serviceContexts =
+            new LinkedBlockingDeque<>();
+
+    /**
+     * Creates a new decorator to capture the {@link ServiceRequestContext}s.
+     */
+    public Function<? super HttpService, HttpService> decorator() {
```

Comment:
Could we also add a variant like this, so that a user can choose what to capture:

```java
    public Function<? super HttpService, HttpService> decorator(Predicate<? super ServiceRequestContext> filter) { ... }
```